pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -335,6 +341,32 @@ func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object
 	})
 }
 
+// deleteSystemProject deletes the system project(s) for a cluster in preparation for deleting the cluster namespace.
+// Normally, the webhook prevents deleting the system project, so Rancher needs to use the sudo user to force it.
+// Otherwise, the deleted namespace will be stuck terminating because it cannot garbage collect the project.
+func (m *mgr) deleteSystemProject(obj runtime.Object, controller string) error {
+	metaAccessor, err := meta.Accessor(obj)
+	if err != nil {
+		return err
+	}
+	projects, err := m.projectLister.List(metaAccessor.GetName(), systemProjectLabels.AsSelector())
+	if err != nil {
+		return err
+	}
+	bypassClient, err := m.projects.WithImpersonation(controllers.WebhookImpersonation())
+	if err != nil {
+		return err
+	}
+	for _, p := range projects {
+		logrus.Infof("[%s] Deleting project %s", controller, p.Name)
+		err = bypassClient.Delete(p.Namespace, p.Name, nil)
+		if err != nil {
+			return err
```

Comment:
Would it be good to return a multi error here as well? Or also check if the error is anything but Not Found?